### ISSUE TITLE
Operationalize and apply triaging issue labels

### ISSUE THREAD

**Issue by tsalo** [nilearn/nilearn#2779]
We have nine new [issue labels](https://github.com/nilearn/nilearn/labels): `priority: low`, `priority: medium`, `priority: high`, `impact: low`, `impact: medium`, `impact: high`, `effort: low`, `effort: medium`, and `effort: high`. These labels, along with the preexisting `good first issue` label, are meant to help triage issues, which the new @nilearn/triaging team will help with.

However, in order to best use these labels, we need to (1) do our best to operationalize them and (2) apply them to existing issues.

Toward the first aim, we need to define what "low", "medium", and "high" effort, priority, and impact should mean. I'll start with the definitions used in [`ME-ICA/tedana`](https://github.com/ME-ICA/tedana/labels), and we can adjust from there:

- effort
   - good first issue: Equivalent to "very low effort"
    - low: Estimated at less than a day's work
    - medium: Estimated at <40h total work
    - high: Estimated at more than 40h total work
- impact
    - low: Improving code/documentation cleanliness/clarity, not function
    - medium: Improves code/documentation functionality for some users
    - high: Enhancement or functionality improvment that will affect most users
- priority
    - low: Issues that are not urgent
    - medium: Should be addressed soon
    - high: Issues that would be really helpful if they were fixed already

**Comment by bthirion** [comment]
Thx for the suggestion.
In the context of the sprint, I believe that the effort task is the most useful. Given the limited time available, I would actually bet that the most important is still to identify effort:low / good first issues.

**Comment by tsalo** [comment]
Absolutely. The immediate concern should be the low-hanging fruit for new contributors. How do folks feel about the definitions though? Are they clear enough? Are there any rules that would make them more useful?

**Comment by bthirion** [comment]
No strong opinion on the alternative. I find  `effort: <low/medium/high>` good enough.

**Comment by jeromedockes** [comment]
Also, should we remove older labels that are redundant with these new ones? for example:
- Good first issue
- Moderate difficulty
- Important

and possibly some of those we (almost) never use such as "hacktoberfest-accepted", "nistats", "in progress"

**Comment by NicolasGensollen** [comment]
@jeromedockes I think "good first issue" should be kept as it is supposed to mean "very low effort". 
I agree that "Moderate difficulty" and "Important" are redundant and that "hacktoberfest-accepted", "nistats", and "in progress" are never used in practice. So, I'm fine with removing them to clarify the list of labels.

**Comment by tsalo** [comment]
"Moderate difficulty" and "Important" were both more prevalent than their newer counterparts, so I ended up just removing the new ones, applying the old ones to any issues where the new ones were used, and renaming the old ones to the new names. 

I haven't touched "hacktoberfest-accepted", "nistats", or "in progress", although I think "nistats" could just be renamed to "GLM" to match other module-specific labels (e.g., "Datasets", "Plotting").

**Comment by bthirion** [comment]
+1 for 'Nistats' -> 'GLM'

**Comment by tsalo** [comment]
Done!

**Comment by tsalo** [comment]
One more label that might be useful would be one to indicate that the issue was a user question and the response was to point the poster to Neurostars. Something like "redirected to neurostars" or "usage"?

**Comment by bthirion** [comment]
Maybe "usage" ?

**Comment by NicolasGensollen** [comment]
+1 for "usage" (maybe "question" but less explicit...)

**Comment by tsalo** [comment]
usage sounds good then. I can create the label and start adding it to any usage questions I see.

The only other thing that I think needs to be covered in this issue is building momentum with adding the effort, impact, and priority labels.

**Comment by tsalo** [comment]
Per today's dev call, we need to add information about these labels and the triage team's responsibilities to the contributing docs. It's probably worth a new issue, but I just wanted to get something written here first in case no one gets around to opening an issue soon.

**Comment by Remi-Gau** [comment]
Follow up on this:

this is the corresponding page in our doc for this: https://nilearn.github.io/stable/maintenance.html#labels

**Comment by Remi-Gau** [comment]
I'd be tempted to add a simple labelling workflow that adds a "needs triage" label to any new issue, so that the triage team knows what issue have not been properly labelled

**Comment by Remi-Gau** [comment]
I have also added a "help wanted" label to flag issues issues that where we are asking for help from contributors or other core-devs. The difference with "good first issue" is that those issues may also require some specific knowledge about the Nilearn code base.

**Comment by tsalo** [comment]
>I'd be tempted to add a simple labelling workflow that adds a "needs triage" label to any new issue, so that the triage team knows what issue have not been properly labelled

I think you can just put it in the `labels` field of the issue templates.

https://github.com/nilearn/nilearn/blob/5c9e42eafb811c04187a2aac01627d12d1024646/.github/ISSUE_TEMPLATE/feature_request.yml?plain=1#L11

https://github.com/nilearn/nilearn/blob/5c9e42eafb811c04187a2aac01627d12d1024646/.github/ISSUE_TEMPLATE/bug.yml?plain=1#L11

**Comment by Remi-Gau** [comment]
Do you know if this can cover blank issues ?

**Comment by tsalo** [comment]
Ah no, I have no clue about blank issues.

**Comment by Remi-Gau** [comment]
brain dump:
- new issues get a "triage needed" label that must be manually removed by a dev (inspired from sklearn) and ensure that an issue has a impact / effort / priority label
- replace "bug" (respectively "enhancement") label with the issue type "bug" (respectively "enhancement") (https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/managing-issue-types-in-an-organization)
- issues opened as bug get the high priority label
- suggestion: maybe replace the 'good first issue' (that seems to attract AI) with our 'Effort; low' label
